src/Dialogs/FingerprintDialog.vala

```diff
@@ -0,0 +1,163 @@
+            pixel_size = 64,
+            margin_start = 12,
+            margin_end = 12,
+            margin_bottom = 12,
+            halign = Gtk.Align.CENTER
+        };
+        form_box.append (fingerprint_image);
+        title_label = new Gtk.Label (_("Enrolling Fingerprint"));
+        title_label.add_css_class (Granite.STYLE_CLASS_H2_LABEL);
+        form_box.append (title_label);
+
+        status_label = new Gtk.Label (_("Touch the fingerprint sensor to begin.")) {
+            wrap = true,
+            justify = Gtk.Justification.CENTER
+        };
+        status_label.add_css_class (Granite.STYLE_CLASS_H3_LABEL);
+        form_box.append (status_label);
+
+        progress_revealer = new Gtk.Revealer () {
+            transition_type = Gtk.RevealerTransitionType.SLIDE_DOWN
+        };
+        form_box.append (progress_revealer);
+
+        cancel_button = add_button (_("Cancel"), Gtk.ResponseType.CANCEL);
+        cancel_button.add_css_class (Granite.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+        try {
+            fp_utils = new SwitchboardPlugUserAccounts.FPUtils ();
+            finish_button = add_button (_("Finish"), Gtk.ResponseType.OK);
+            finish_button.add_css_class (Granite.STYLE_CLASS_SUGGESTED_ACTION);
+            finish_button.sensitive = false;
+        } catch (Error e) {
+            warning ("Failed to initialize fingerprint device: %s".printf (e.message));
+            status_label.label = _("Fingerprint device not available");
+            cancel_button.sensitive = true;
+            return;
+        }
+
+
+        response.connect ((response_id) => {
+            destroy ();
+        });
+
+        FPUtils.device.enroll_status.connect (set_progress);
+
+        this.unmap.connect (() => {
+            fp_utils.enroll_stop ();
+            fp_utils.release ();
+            FPUtils.device.enroll_status.disconnect (set_progress);
+        });
+
+        if (fp_utils.claim ()) {
+            fp_utils.enroll_start_async.begin ((obj, res) => {
+                bool success = fp_utils.enroll_start_async.end (res);
+                if (success) {
+                    progress_bar = new Gtk.LevelBar () {
+                        min_value = 1,
+                        max_value = FPUtils.device.num_enroll_stages + 1,
+                        value = 1,
+                        margin_top = 12,
+                        mode = DISCRETE
+                    };
+                    progress_revealer.child = progress_bar;
+                } else {
+                    destroy ();
+                }
+            });
+        }
+    }
+
+    private void set_progress (string message, bool done) {
+        status_label.label = get_friendly_status (message);
+        if (done) {
+            status_label.label = _("You are all set!");
+            cancel_button.sensitive = false;
+            finish_button.sensitive = true;
+            finish_button.grab_focus ();
```

Comment:
Instead of grab focus this should be set as default for the window https://valadoc.org/gtk4/Gtk.Window.set_default_widget.html